scipy/stats/stats.py

```diff
@@ -6134,9 +6145,25 @@ def power_divergence(f_obs, f_exp=None, ddof=0, axis=0, lambda_=None):
         lambda_ = 1
 
     f_obs = np.asanyarray(f_obs)
+    f_obs_float = f_obs.astype(np.float64)
 
     if f_exp is not None:
-        f_exp = np.asanyarray(f_exp)
+        f_exp = _m_broadcast_to(np.asanyarray(f_exp), f_obs_float.shape)
```

Comment:
With this change, the test and the refguide-check pass:
```suggestion
        f_exp = np.asanyarray(f_exp)
        bshape = _broadcast_shapes(f_obs_float.shape, f_exp.shape)
        f_obs_float = _m_broadcast_to(f_obs_float, bshape)
        f_exp = _m_broadcast_to(f_exp, bshape)
```
This figures out the common broadcast shape, and then uses `_m_broadcast_to()` on both inputs.

`_broadcast_shapes()` is already defined here in `stats.py`.

What do you think?

Reply:
It makes me cry a little.

> _broadcast_shapes() is already defined here in stats.py.
```
np.broadcast(x, y).shape
```

But I'll commit it.

Reply:
Looks like that would work fine too, if you prefer it.